crates/solver/src/interactions/allowances.rs

```diff
@@ -397,27 +389,21 @@ mod tests {
         let web3 = mock::web3();
         web3.transport()
             .mock()
-            .expect_execute_batch()
-            .returning(move |calls| {
-                Ok(calls
-                    .into_iter()
-                    .map(|(method, params)| {
-                        assert_eq!(method, "eth_call");
-
-                        let call =
-                            serde_json::from_value::<CallRequest>(params[0].clone()).unwrap();
-                        assert_eq!(call.data.unwrap(), allowance_call_data(owner, spender));
-
-                        let to = call.to.unwrap();
-                        if to == addr!("1111111111111111111111111111111111111111") {
-                            Ok(allowance_return_data(1337.into()))
-                        } else if to == addr!("2222222222222222222222222222222222222222") {
-                            Err(web3::Error::Decoder("test error".to_string()))
-                        } else {
-                            panic!("call to unexpected token {to:?}")
-                        }
-                    })
-                    .collect())
+            .expect_execute()
```

Comment:
The web3 instance we use for real auto batches requests under the hood such that all requests can be made with `eth_call` instead.